plenum/test/primary_selection/test_recover_more_than_f_failure.py

```diff
@@ -44,21 +40,71 @@ def test_recover_stop_primaries(looper, txnPoolNodeSet, allPluginsPath, tconf, c
     ensureElectionsDone(looper=looper, nodes=active_nodes, numInstances=2)
     ensure_all_nodes_have_same_data(looper, nodes=active_nodes)
 
-    # send request
-    sendReqsToNodesAndVerifySuffReplies(looper, wallet1, client1, numReqs=10)
+    # send at least one checkpoint
+    assert nodes_do_not_have_checkpoints(*active_nodes)
+    sendReqsToNodesAndVerifySuffReplies(looper, wallet1, client1, numReqs=2*checkpoint_size)
+    assert nodes_have_checkpoints(*active_nodes)
     ensure_all_nodes_have_same_data(looper, nodes=active_nodes)
 
-    # Stop second node (current Primary)
+    # Stop second node (current Primary) so the primary looses his state
     stopped_node, active_nodes = stop_primary(looper, active_nodes)
 
-    # Restart second node
+    # Restart the primary node
     restarted_node = start_stopped_node(stopped_node, looper, tconf, stopped_node.basedirpath, allPluginsPath)
+    assert nodes_do_not_have_checkpoints(restarted_node)
+    assert nodes_have_checkpoints(*active_nodes)
     active_nodes = active_nodes + [restarted_node]
 
     # Check that primary selected
     ensureElectionsDone(looper=looper, nodes=active_nodes, numInstances=2, customTimeout=30)
     waitForViewChange(looper, active_nodes, expectedViewNo=expected_view_no)
     ensure_all_nodes_have_same_data(looper, nodes=active_nodes)
 
-    sendReqsToNodesAndVerifySuffReplies(looper, wallet1, client1, numReqs=10)
+    # Check if the pool is able to process requests
+    sendReqsToNodesAndVerifySuffReplies(looper, wallet1, client1, numReqs=10*checkpoint_size)
     ensure_all_nodes_have_same_data(looper, nodes=active_nodes)
```

Comment:
Shouldn't we check that checkoints were sent. For example, call `nodes_have_checkpoints(*active_nodes)` at the end?